src/caffe/proto/caffe.proto

```diff
@@ -2,13 +2,21 @@ syntax = "proto2";
 
 package caffe;
 
+// Specifies the shape (dimensions) of a Blob.
+message BlobShape {
+  repeated int64 dim = 1 [packed = true];
```

Comment:
But Jeff, there is no reason for any individual to have a > 2GB blob in their home. Haha no, I'm all for `int64` dimensions.